### PR TITLE
Fix all bugs on login and logout

### PR DESCRIPTION
**Pull request by vbertoletti**
This PR fixes bugs on login screen, profile screen and realm/user provider, as they are all connected to login and logout. Profile page renders data from db without errors.

### CODE REVIEW

File: client/js/screens/LogIn/LogInContainer.js [hunk 1]
```diff
@@ -22,11 +22,13 @@ class LoginContainer extends Component {
       <Mutation mutation={LoginMutation}>
         {login => (
           <UserContext.Consumer>
-            {({ signin, token }) => {
-              if (token) {
+            {({ storeSessionToken, token }) => {
+              if (token.length > 0) {
```
Comment (siddharthparmar7) [inline client/js/screens/LogIn/LogInContainer.js]: if you make your Context state null initially you will not have to check for your token's length. You can just do `if(token)`

File: client/js/screens/LogIn/LogInContainer.js [hunk 2]
```diff
@@ -22,11 +22,13 @@ class LoginContainer extends Component {
...
                 this.props.navigation.navigate("App");
               } else {
-                return <LogIn login={login} signin={signin} />;
+                return (
+                  <LogIn login={login} storeSessionToken={storeSessionToken} />
+                );
```
Comment (siddharthparmar7) [inline client/js/screens/LogIn/LogInContainer.js:32]: add prop type validation.

File: client/js/screens/Profile/ProfileContainer.js [hunk 3]
```diff
@@ -21,7 +35,39 @@ class ProfileContainer extends Component {
...
+                      />
+                    );
+                  }
+                }}
+              </Query>
+            );
+          }
+        }}
+      </UserContext.Consumer>
+    );
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/ProfileContainer.js:71]: please add prop type validations.

File: client/js/screens/Profile/Profile.js [hunk 4]
```diff
@@ -51,11 +57,12 @@ const Profile = ({ navigation }) => {
...
+        <FontAwesome5 name={"chevron-right"} size={20} solid />
       </TouchableOpacity>
       <TouchableOpacity
         style={styles.profileButton}
         onPress={() => {
-          navigation.navigate("Login");
+          logout(id);
         }}
       >
         <Text style={styles.buttonText}>LOGOUT</Text>
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/Profile.js:69]: add prop type validation.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
I can not check your app because it throws an error Unable to resolve module `../screens/Login` from root stack navigation. Please fix it. Also, you should be calling your dirs, files and components SignIn and SignUp, not LogIn and SignUp. Please make the changes. Thank you.

**Comment by vbertoletti** [comment]
Fixed all requirements, thanks.

File: client/js/context/UserContext/UserProvider.js [hunk 5]
```diff
@@ -6,34 +6,44 @@ class UserProvider extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      token: "Test Token",
-      token: "",
+      token: null,
       id: ""
```
Comment (siddharthparmar7) [inline client/js/context/UserContext/UserProvider.js]: make this null too.

File: client/js/context/UserContext/UserProvider.js [hunk 6]
```diff
@@ -6,34 +6,44 @@ class UserProvider extends Component {
...
 
   async removeUserIdToken(id) {
-    await Realm.write(() => {
-      Realm.delete(Realm.objectForPrimaryKey("User", id));
-      this.setState({ id: "", token: "" });
-    });
+    try {
+      await Realm.write(() => {
+        Realm.delete(Realm.objectForPrimaryKey("User", id));
+        this.setState({ id: "", token: null });
```
Comment (siddharthparmar7) [inline client/js/context/UserContext/UserProvider.js]: id should be null too.

File: client/js/screens/Profile/Profile.js [hunk 7]
```diff
@@ -64,3 +72,10 @@ const Profile = ({ navigation }) => {
   );
 };
 export default Profile;
+
+Profile.propTypes = {
+  navigation: PropTypes.object,
+  data: PropTypes.object,
+  logout: PropTypes.func,
+  id: PropTypes.string
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/Profile.js]: the props that are not required, you should have default props for them.

File: client/js/screens/Profile/ProfileContainer.js (hunk 3 above)
Comment (siddharthparmar7) [reply client/js/screens/Profile/ProfileContainer.js:71]: Please resolve this.

File: client/js/screens/SignIn/SignIn.js [hunk 8]
```diff
@@ -10,10 +10,11 @@ import {
...
 import styles from "./styles";
+import PropTypes from "prop-types";
 
 const required = value => (value ? undefined : "* Required Field");
 
-const LogIn = ({ login, storeSessionToken }) => {
+const SignIn = ({ login, storeSessionToken }) => {
   return (
     <Fragment>
       <StatusBar barStyle="light-content" />
```
Comment (siddharthparmar7) [inline client/js/screens/SignIn/SignIn.js:21]: make all your View components Fragment that is just by themselves.

File: client/js/screens/SignIn/SignIn.js [hunk 9]
```diff
@@ -111,4 +112,9 @@ const LogIn = ({ login, storeSessionToken }) => {
   );
 };
 
-export default LogIn;
+export default SignIn;
+
+SignIn.propTypes = {
+  login: PropTypes.func,
+  storeSessionToken: PropTypes.func
```
Comment (siddharthparmar7) [inline client/js/screens/SignIn/SignIn.js]: add default props if these props are not required.

File: client/js/screens/LogIn/LogInContainer.js (hunk 2 above)
Comment (siddharthparmar7) [reply client/js/screens/LogIn/LogInContainer.js:32]: this is not resolved yet.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please resolved all the issues. Thank you.

**Comment by vbertoletti** [comment]
Amended proptypes as required, id initial state is now null, also added fragment to one section that was wrapped in unnecessary view tag, fixed LogInContainer to be SignInContainer.

File: client/js/context/UserContext/UserProvider.js (hunk 6 above)
Comment (siddharthparmar7) [reply client/js/context/UserContext/UserProvider.js]: ID is not null here.

File: client/js/screens/Profile/ProfileContainer.js [hunk 10]
```diff
@@ -21,8 +35,46 @@ class ProfileContainer extends Component {
...
   }
 }
 
 export default ProfileContainer;
+
+ProfileContainer.defaultProps = {
+  removeUserIdToken: null,
+  id: null,
+  data: null
```
Comment (siddharthparmar7) [inline client/js/screens/Profile/ProfileContainer.js]: these are not props. They are function arguments. 

the only prop here is navigation.

File: client/js/screens/SignIn/SignInContainer.js [hunk 11]
```diff
@@ -36,4 +38,10 @@ class LoginContainer extends Component {
   }
 }
 
-export default LoginContainer;
+export default SignInContainer;
+
+SignInContainer.defaultProps = {
+  login: null,
+  storeSessionToken: null,
+  token: null
```
Comment (siddharthparmar7) [inline client/js/screens/SignIn/SignInContainer.js]: these are not props. these are function arguments.

You should have all of your prop validations with .isRequired and if they are not required only then have default props.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please make the requested changes and don't validate your function arguments. Validate your props.

**Comment by vbertoletti** [comment]
Thanks for taking the time to clarify, I have pushed more changes. Please let me know if something needs improvements.

**Review by siddharthparmar7 — APPROVED** [review]
Thank you!

**Pull request merged**